### PR TITLE
Add console checks for sct divergence

### PR DESCRIPTION
**Pull request by grod220**
Closes https://github.com/penumbra-zone/web/issues/878

https://github.com/penumbra-zone/web/assets/16624263/c0596976-bc1f-45f9-8bc6-a84e7aaec7fa

### CODE REVIEW

File: packages/query/src/block-processor.ts [hunk 1]
```diff
@@ -141,7 +148,7 @@ export class BlockProcessor implements BlockProcessorInterface {
 
   private async syncAndStore() {
     const fullSyncHeight = await this.indexedDb.getFullSyncHeight();
-    const startHeight = fullSyncHeight ? fullSyncHeight + 1n : 0n;
+    const startHeight = fullSyncHeight !== undefined ? fullSyncHeight + 1n : 0n; // Must compare to undefined as 0n is falsy
```
Comment (grod220) [inline packages/query/src/block-processor.ts:151]: Found an interesting bug. We sync block 0. However, 0 is a falsy value. That means if state was reset between blocks 0 and 1000, we'd have a sct root divergence.

Comment (Valentine1898) [reply packages/query/src/block-processor.ts:151]: Nice catch 👍 
I wonder if this fixes the divergence that occurs on Henry's wallet (I suspect Henry didn't stop synchronization specifically between the 0 and 1000 block)

**Review by Valentine1898 — APPROVED** [review]

File: packages/query/src/block-processor.ts (hunk 1 above)
Comment (hdevalence) [reply packages/query/src/block-processor.ts:151]: Hmm, I _did_ make transactions in the first 1000 blocks, so if it paused to fetch full tx details, that could have been an opportunity for interruption...

File: packages/query/src/block-processor.ts [hunk 2]
```diff
@@ -36,6 +36,13 @@ import { getAssetId } from '@penumbra-zone/getters/src/metadata';
 import { STAKING_TOKEN_METADATA } from '@penumbra-zone/constants/src/assets';
 import { toDecimalExchangeRate } from '@penumbra-zone/types/src/amount';
 import { ValidatorInfoResponse } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/stake/v1/stake_pb';
+import { uint8ArrayToHex } from '@penumbra-zone/types/src/hex';
+
+declare global {
+  // `var` required for global declaration (as let/const are block scoped)
+  // eslint-disable-next-line no-var
+  var ASSERT_ROOT_VALID: boolean | undefined;
+}
```
Comment (hdevalence) [inline packages/query/src/block-processor.ts:45]: @grod220 can we get some docs in the dev docs about the existence of this global and how you would set it and why?

Comment (grod220) [reply packages/query/src/block-processor.ts:45]: Sounds good 👍 
https://github.com/penumbra-zone/web/pull/891

**Pull request merged**